### ISSUE TITLE
Alias is not used by the _find for arguments parsing

### ISSUE THREAD

**Issue by rubalo** [admiralobvious/vyper#17]
When looking for an argument, the _find function doesn't search for an alias registered

```python
import vyper
import argparse
import os

v = vyper.Vyper()

v.automatic_env()
v.register_alias('arg', 'name')
v.bind_env('name', 'ENV_VAR')
v.set_default('name', 'default_val')

ap = argparse.ArgumentParser()
ap.add_argument('--arg', type=str)
args = vars(ap.parse_args(['--arg', 'arg_val']))

v.bind_args(args)

print(v.get('name'))
os.environ['ENV_VAR'] = 'env_val'
print(v.get('name'))

# Result
#   default_val
#   env_val
# Expected result
#   arg_val
#   arg_val
#   Because Overrides > Args > Env variables > Config files > Defaults
```

Even worst, the Argparse binding mess up the default parameter

```python
import vyper
import argparse

v = vyper.Vyper()
v.automatic_env()

ap = argparse.ArgumentParser()
ap.add_argument('--arg', type=str)

v.register_alias('arg', 'name')
v.set_default('name', 'default_val')
v.bind_args(ap)

print(v.get('name'))

# Result
#   None
# Expected Result
#   default_val
```

**Comment by admiralobvious** [comment]
Indeed, the changes I did to the binding of arguments, may have had some unintended consequences. I'll take a look at the first issue ASAP.

The second issue you point out is "normal" though because you didn't specify a default when you added the argument`ap.add_argument('--arg', type=str)`. If you set the default after `bind_args()` it should work as intended. I had already added the following to the `README.md` explaining the change in behaviour of `bind_args()` in relation to defaults:
> The method also sets defaults based on what you pass via add_argument() default parameter.
> Note: If you don't specify a default, the values will be set to None.

It was not in the release notes though, while it should've been.

Also, while personally in my work/projects I do use arguments as the defaults, maybe it should not be the default behaviour of `bind_args()` and/or maybe it should have a parameter to disable/enable that behaviour.

**Comment by rubalo** [comment]
Thanks for your answer ! It s quite difficult to anticipate the behavior since a lot of things depends on the order in the vyper initialization. Another example of this is:
```python
import vyper
import os

v = vyper.Vyper()


os.environ["TEMP_VAR"] = "original"
v.automatic_env()
v.bind_env("test", "TEMP_VAR")
assert "original" == v.get("test")
v.set("test", "modified")
assert "modified" == v.get("test")
os.environ["TEMP_VAR"] = "updated"
assert "updated" == v.get("test")


# Traceback (most recent call last):
#   File "/home/jbarbou/git_local/slack/bots/rodrigo/rodrigo/test.py", line 14, in <module>
#     assert "updated" == v.get("test")
# AssertionError
```

Seems that the register_alias is still under construction :-) the quotes between the "alias"shouldn't be there I guess.

IMHO, it could be good to flatten the config before returning the values as it is done in the go lib but I'm not an expert in go so I might get the wrong idea.

Anyway thanks again !

**Comment by admiralobvious** [comment]
>Seems that the register_alias is still under construction :-) the quotes between the "alias"shouldn't be there I guess.

Yeah, that should've been there, don't know why it was like that, but I fixed that in 67d1507d1c953357f1e5912b603ecaf99228e3c5.

I am working on the other issues.

**Comment by rubalo** [comment]
Thanks for looking into this, it is a big tricky but the debug function helps a lot !
